File: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/examples-java/Workflows_ListBySubscription.java
/** Samples for Workflows List. */
public final class Main {
    /*
     * x-ms-original-file: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/examples/Workflows_ListBySubscription.json
     */
    /**
     * Sample code: List all workflows in a subscription.
     *
     * @param manager Entry point to LogicManager.
     */
    public static void listAllWorkflowsInASubscription(com.azure.resourcemanager.logic.LogicManager manager) {
        manager.workflows().list(null, null, com.azure.core.util.Context.NONE);
    }
}

File: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/examples-js/Workflows_ListBySubscription.js
const { LogicManagementClient } = require("@azure/arm-logic");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Gets a list of workflows by subscription.
 *
 * @summary Gets a list of workflows by subscription.
 * x-ms-original-file: specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/examples/Workflows_ListBySubscription.json
 */
async function listAllWorkflowsInASubscription() {
  const subscriptionId =
    process.env["LOGIC_SUBSCRIPTION_ID"] || "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345";
  const credential = new DefaultAzureCredential();
  const client = new LogicManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.workflows.listBySubscription()) {
    resArray.push(item);
  }
  console.log(resArray);
}
